src/reducers/fileTree.tsx

```diff
@@ -191,6 +190,90 @@ export const getRelativePath = ({
   return pathList[newIndex];
 };
 
+const findRelativeMessageUid = (
+  currentPath: string,
+  messageMap: LinterMessageMap,
+  pathList: string[],
+  position: RelativePathPosition,
+): LinterMessage['uid'] | null => {
+  let path = currentPath;
+  const maxAttempts = pathList.length;
+  for (let attempt = 1; attempt <= maxAttempts; attempt++) {
+    const nextPath = getRelativePath({
+      currentPath: path,
+      pathList,
+      position,
+    });
+    const nextMessages = messageMap[nextPath];
```

Comment:
I shortened them a bit, but I'm not sure what you're looking for. Did you want things as short as `const m = getMessagesForPath(nextMessages)`? I try to avoid things like that. Take a look at the new version (once I push it) and see if it's short enough for you.